app/models/order.rb

```diff
@@ -11,4 +11,13 @@ def grandtotal
   def total_quantity
     item_orders.sum(:quantity)
   end
+
+  def cancel
+    update(status: "Cancelled")
+
+    item_orders.each do |item_order|
+      item_order.update(status: 0)
+      item_order.item.update(inventory: item_order.item.inventory + item_order.quantity)
+    end
```

Comment:
This entire method takes care of the aftermath of cancellation.
By US30, there's nothing that's present to change an order to fulfilled but in the future, it will take care of that.
Inventory adds the quantity- not a method present that reduces the quantity when an item has been purchased.

Reply:
Great helper method, this accomplishes a lot within a single method